test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -63,6 +68,34 @@
           (is (= "something-crazy/for-content-type"
                  (get-in resp [:request :content-type]))))))))
 
+(deftest code-id-injection-test
+  (testing "code_id is calculated and added to the catalog request."
+    (let [handler (fn ([req] {:request req}))
+          app     (build-ring-handler handler "1.2.3" (constantly "foo-is-my-codeid"))
+          resp    (app {:request-method :get
+                        :uri "/v3/catalog/bar"})]
+      (is (= "foo-is-my-codeid" (get-in resp [:request :params "code_id"])))))
+  (testing "code_id is not added to non-catalog requests"
+    (let [handler (fn ([req] {:request req}))
+          app (build-ring-handler handler "1.2.3" (constantly "foo-is-my-codeid"))
+          request     (partial app-request app)]
```

Comment:
Not important, but the let alignment on this and the `testing` block above are inconsistent. I think we're not vertically aligning them?